<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.2.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">



<link rel="stylesheet" href="//cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.14.0/css/all.min.css">
  <link rel="stylesheet" href="//cdn.jsdelivr.net/npm/animate.css@3.1.1/animate.min.css">


<script class="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"example.com","root":"/","scheme":"Gemini","version":"8.0.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12},"copycode":false,"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"fadeInDown","post_body":"fadeInDown","coll_header":"fadeInLeft","sidebar":"fadeInUp"}},"prism":false,"i18n":{"placeholder":"Searching...","empty":"We didn't find any results for the search: ${query}","hits_time":"${hits} results found in ${time} ms","hits":"${hits} results found"},"path":"search.xml","localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false}};
  </script>

  <meta name="description" content="简介edgeR作用对象是count文件，rows 代表基因，行代表文库，count代表的是比对到每个基因的reads数目。它主要关注的是差异表达分析，而不是定量基因表达水平。 edgeR works on a table of integer read counts, with rows corresponding to genes and columns to independent libra">
<meta property="og:type" content="article">
<meta property="og:title" content="RNAseq_6_无重复差异基因分析">
<meta property="og:url" content="http://example.com/2020/11/23/RNAseq-6-%E6%97%A0%E9%87%8D%E5%A4%8D%E5%B7%AE%E5%BC%82%E5%9F%BA%E5%9B%A0%E5%88%86%E6%9E%90/index.html">
<meta property="og:site_name" content="WENJU QIAN‘s tech blog">
<meta property="og:description" content="简介edgeR作用对象是count文件，rows 代表基因，行代表文库，count代表的是比对到每个基因的reads数目。它主要关注的是差异表达分析，而不是定量基因表达水平。 edgeR works on a table of integer read counts, with rows corresponding to genes and columns to independent libra">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/6_1.jpg">
<meta property="og:image" content="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/6_2.jpg">
<meta property="og:image" content="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/6_3.jpg">
<meta property="og:image" content="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/6_4.jpg">
<meta property="og:image" content="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/6_5.jpg">
<meta property="og:image" content="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/6_6.jpg">
<meta property="og:image" content="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/6_7.jpg">
<meta property="article:published_time" content="2020-11-22T16:25:57.000Z">
<meta property="article:modified_time" content="2020-11-30T16:02:17.327Z">
<meta property="article:author" content="WENJU QIAN">
<meta property="article:tag" content="RNAseq">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/6_1.jpg">


<link rel="canonical" href="http://example.com/2020/11/23/RNAseq-6-%E6%97%A0%E9%87%8D%E5%A4%8D%E5%B7%AE%E5%BC%82%E5%9F%BA%E5%9B%A0%E5%88%86%E6%9E%90/">


<script class="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'en'
  };
</script>

  <title>RNAseq_6_无重复差异基因分析 | WENJU QIAN‘s tech blog</title>
  






  <noscript>
  <style>
  body { margin-top: 2rem; }

  .use-motion .menu-item,
  .use-motion .sidebar,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header {
    visibility: visible;
  }

  .use-motion .header,
  .use-motion .site-brand-container .toggle,
  .use-motion .footer { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle,
  .use-motion .custom-logo-image {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line {
    transform: scaleX(1);
  }

  .search-pop-overlay, .sidebar-nav { display: none; }
  .sidebar-panel { display: block; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage" class="use-motion">
  <div class="headband"></div>

  <main class="main">
    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
        <span class="toggle-line"></span>
        <span class="toggle-line"></span>
        <span class="toggle-line"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <i class="logo-line"></i>
      <h1 class="site-title">WENJU QIAN‘s tech blog</h1>
      <i class="logo-line"></i>
    </a>
      <p class="site-subtitle" itemprop="description">时光匆匆独白，将颠沛磨成卡带</p>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>



<nav class="site-nav">
  <ul class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="/about/" rel="section"><i class="fa fa-user fa-fw"></i>About</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/tags/" rel="section"><i class="fa fa-tags fa-fw"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/categories/" rel="section"><i class="fa fa-th fa-fw"></i>Categories</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a>

  </li>
        <li class="menu-item menu-item-schedule">

    <a href="/schedule/" rel="section"><i class="fa fa-calendar fa-fw"></i>Schedule</a>

  </li>
        <li class="menu-item menu-item-sitemap">

    <a href="/sitemap.xml" rel="section"><i class="fa fa-sitemap fa-fw"></i>Sitemap</a>

  </li>
        <li class="menu-item menu-item-commonweal">

    <a href="/404/" rel="section"><i class="fa fa-heartbeat fa-fw"></i>Commonweal 404</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off" maxlength="80"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div class="search-result-container no-result">
  <div class="search-result-icon">
    <i class="fa fa-spinner fa-pulse fa-5x"></i>
  </div>
</div>

    </div>
  </div>

</div>
        
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line"></span>
    <span class="toggle-line"></span>
    <span class="toggle-line"></span>
  </div>

  <aside class="sidebar">

    <div class="sidebar-inner sidebar-nav-active sidebar-toc-active">
      <ul class="sidebar-nav">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <section class="post-toc-wrap sidebar-panel">
          <div class="post-toc animated"><ol class="nav"><li class="nav-item nav-level-2"><a class="nav-link" href="#%E7%AE%80%E4%BB%8B"><span class="nav-number">1.</span> <span class="nav-text">简介</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#1-%E9%A6%96%E5%85%88%E5%AE%89%E8%A3%85edgeR-%E5%8C%85"><span class="nav-number">2.</span> <span class="nav-text">1) 首先安装edgeR 包</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#2%EF%BC%89%E7%9F%A9%E9%98%B5%E6%9E%84%E5%BB%BA%E5%8F%8A%E5%B7%AE%E5%BC%82%E5%88%86%E6%9E%90"><span class="nav-number">3.</span> <span class="nav-text">2）矩阵构建及差异分析</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#2-1-%E8%A1%A8%E8%BE%BE%E7%9F%A9%E9%98%B5"><span class="nav-number">3.1.</span> <span class="nav-text">2.1) 表达矩阵</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-2-%E6%8E%A5%E7%9D%80%E6%9E%84%E5%BB%BADGEList%E5%AF%B9%E8%B1%A1"><span class="nav-number">3.2.</span> <span class="nav-text">2.2 接着构建DGEList对象</span></a><ol class="nav-child"><li class="nav-item nav-level-5"><a class="nav-link" href="#%E6%95%B0%E6%8D%AE%E8%BF%87%E6%BB%A4"><span class="nav-number">3.2.0.1.</span> <span class="nav-text">数据过滤</span></a></li></ol></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E5%B7%AE%E5%BC%82%E8%A1%A8%E8%BE%BE%E5%88%86%E6%9E%90"><span class="nav-number">3.3.</span> <span class="nav-text">差异表达分析</span></a></li></ol></li></ol></div>
      </section>
      <!--/noindex-->

      <section class="site-overview-wrap sidebar-panel">
        <div class="site-author animated" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="WENJU QIAN"
      src="/images/avatar.jpg">
  <p class="site-author-name" itemprop="name">WENJU QIAN</p>
  <div class="site-description" itemprop="description">Student From Shanghai Institute of Nutrition and Health, CAS</div>
</div>
<div class="site-state-wrap animated">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">19</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
          
        <span class="site-state-item-count">3</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author animated">
      <span class="links-of-author-item">
        <a href="https://github.com/qianwenju" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;qianwenju" rel="noopener" target="_blank"><i class="fab fa-github fa-fw"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:qianwenju2018@sibs.ac.cn" title="E-Mail → mailto:qianwenju2018@sibs.ac.cn" rel="noopener" target="_blank"><i class="fa fa-envelope fa-fw"></i>E-Mail</a>
      </span>
  </div>



      </section>
    </div>
  </aside>
  <div class="sidebar-dimmer"></div>
 


    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>

<noscript>
  <div class="noscript-warning">Theme NexT works best with JavaScript enabled</div>
</noscript>


    <div class="main-inner post posts-expand">
      

      

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="http://example.com/2020/11/23/RNAseq-6-%E6%97%A0%E9%87%8D%E5%A4%8D%E5%B7%AE%E5%BC%82%E5%9F%BA%E5%9B%A0%E5%88%86%E6%9E%90/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.jpg">
      <meta itemprop="name" content="WENJU QIAN">
      <meta itemprop="description" content="Student From Shanghai Institute of Nutrition and Health, CAS">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="WENJU QIAN‘s tech blog">
    </span>

    
    
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          RNAseq_6_无重复差异基因分析
        </h1>

        <div class="post-meta-container">
          <div class="post-meta">
    <span class="post-meta-item">
      <span class="post-meta-item-icon">
        <i class="far fa-calendar"></i>
      </span>
      <span class="post-meta-item-text">Posted on</span>

      <time title="Created: 2020-11-23 00:25:57" itemprop="dateCreated datePublished" datetime="2020-11-23T00:25:57+08:00">2020-11-23</time>
    </span>
      <span class="post-meta-item">
        <span class="post-meta-item-icon">
          <i class="far fa-calendar-check"></i>
        </span>
        <span class="post-meta-item-text">Edited on</span>
        <time title="Modified: 2020-12-01 00:02:17" itemprop="dateModified" datetime="2020-12-01T00:02:17+08:00">2020-12-01</time>
      </span>

  
  
  <span class="post-meta-item">
    
      <span class="post-meta-item-icon">
        <i class="far fa-comment"></i>
      </span>
      <span class="post-meta-item-text">Valine: </span>
    
    <a title="valine" href="/2020/11/23/RNAseq-6-%E6%97%A0%E9%87%8D%E5%A4%8D%E5%B7%AE%E5%BC%82%E5%9F%BA%E5%9B%A0%E5%88%86%E6%9E%90/#valine-comments" itemprop="discussionUrl">
      <span class="post-comments-count valine-comment-count" data-xid="/2020/11/23/RNAseq-6-%E6%97%A0%E9%87%8D%E5%A4%8D%E5%B7%AE%E5%BC%82%E5%9F%BA%E5%9B%A0%E5%88%86%E6%9E%90/" itemprop="commentCount"></span>
    </a>
  </span>
  
  
</div>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">
        <h2 id="简介"><a href="#简介" class="headerlink" title="简介"></a>简介</h2><p>edgeR作用对象是count文件，rows 代表基因，行代表文库，count代表的是比对到每个基因的reads数目。它主要关注的是差异表达分析，而不是定量基因表达水平。</p>
<p>edgeR works on a table of integer read counts, with rows corresponding to genes and columns to independent libraries. The counts represent the total number of reads aligning to each gene (or other genomic locus).edgeR is concerned with diﬀerential expression analysis rather than with the quantiﬁcation of expression levels. It is concerned with relative changes in expression levels between conditions,but not directly with estimating absolute expression levels.</p>
<p>edgeR作用的是真实的比对统计，因此不建议用预测的转录本</p>
<p>Note that edgeR is designed to work with actual read counts. We not recommend that predicted transcript abundances are input the edgeR in place of actual counts.</p>
<p><strong>归一化原因：</strong></p>
<blockquote>
<p>技术原因影响差异表达分析：<br>1）Sequencing depth：统计测序深度(即代表的是library size)；<br>2）RNA composition：个别异常高表达基因导致其它基因采样不足<br>3）GC content： sample-speciﬁc eﬀects for GC-content can be detected<br>4）sample-speciﬁc eﬀects for gene length have been detected</p>
</blockquote>
<p><strong>注意：edgeR必须是原始表达量，而不能是rpkm等矫正过的。</strong><br>Note that normalization in edgeR is model-based, and the original read counts are not themselves transformed. This means that users should not transform the read counts in any way before inputing them to edgeR. For example, users should not enter RPKM or FPKM values to edgeR in place of read counts. Such quantities will prevent edgeR from correctly estimating the mean-variance relationship in the data, which is a crucial to the statistical strategies underlying edgeR.Similarly, users should not add artiﬁcial values to the counts before inputing them to edgeR.<br>个人是不太推荐没有重复的差异表达分析，因为毕竟统计学上的p值是为了证明两个样本的差异是真实存在而不是抽样误差导致的，<br>因此每当别人提问的时候, 我个人的建议就是定性看看倍数变化吧. 但是如果真的强行要算p值, 其实也不是不行, edgeR就是一种选择.</p>
<h2 id="1-首先安装edgeR-包"><a href="#1-首先安装edgeR-包" class="headerlink" title="1) 首先安装edgeR 包"></a>1) 首先安装edgeR 包</h2><figure class="highlight plain"><table><tr><td class="code"><pre><span class="line">#如果没有安装BiocMaRnager则先安装BiocManager，之后通过BiocManager安装edgeR包</span><br><span class="line"></span><br><span class="line">if (!requireNamespace(&quot;BiocManager&quot;, quietly &#x3D; TRUE))</span><br><span class="line">    install.packages(&quot;BiocManager&quot;)</span><br><span class="line"></span><br><span class="line">BiocManager::install(&quot;edgeR&quot;)</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p>安装结束之后开始处理文件</p>
<h2 id="2）矩阵构建及差异分析"><a href="#2）矩阵构建及差异分析" class="headerlink" title="2）矩阵构建及差异分析"></a>2）矩阵构建及差异分析</h2><p>需要构建2个矩阵：1、表达矩阵；2、分组矩阵( 实验设计)；</p>
<h3 id="2-1-表达矩阵"><a href="#2-1-表达矩阵" class="headerlink" title="2.1) 表达矩阵"></a>2.1) 表达矩阵</h3><p>读取文件</p>
<figure class="highlight plain"><table><tr><td class="code"><pre><span class="line"># 首先读取counts文件之后查看count文件前6行</span><br><span class="line">&gt; rawdata &lt;- read.csv(file &#x3D; &quot;C:&#x2F;&#x2F;Users&#x2F;My&#x2F;Desktop&#x2F;diff_name&quot;,header &#x3D; T,stringsAsFactors &#x3D; F)</span><br><span class="line"># 查看读取的diff_name文件</span><br><span class="line">&gt; head(rawdata)</span><br><span class="line">  X    ensembl_gene_id               gene_id control_W55 test_K54 external_gene_name</span><br><span class="line">1 1 ENSMUSG00000000001  ENSMUSG00000000001.4           7       11              Gnai3</span><br><span class="line">2 2 ENSMUSG00000000003 ENSMUSG00000000003.15           0        0               Pbsn</span><br><span class="line">3 3 ENSMUSG00000000028 ENSMUSG00000000028.15           1        0              Cdc45</span><br><span class="line">4 4 ENSMUSG00000000031 ENSMUSG00000000031.16           0        2                H19</span><br><span class="line">5 5 ENSMUSG00000000037 ENSMUSG00000000037.17           0        0              Scml2</span><br><span class="line">6 6 ENSMUSG00000000049 ENSMUSG00000000049.11          54       33               Apoh</span><br></pre></td></tr></table></figure>

<p>读取完数据之后我们先预处理一下数据，比如我只想要ensembl_gene_id、control_w55、tese_k54、external_gene_name这几列，并调整一下顺序。</p>
<figure class="highlight plain"><table><tr><td class="code"><pre><span class="line">&gt; swap_rawdata &lt;- cbind(rawdata$ensembl_gene_id,rawdata$external_gene_name,rawdata$control_W55,rawdata$test_K54)</span><br><span class="line">&gt; head(swap_rawdata)</span><br><span class="line">     [,1]                 [,2]    [,3] [,4]</span><br><span class="line">[1,] &quot;ENSMUSG00000000001&quot; &quot;Gnai3&quot; &quot;7&quot;  &quot;11&quot;</span><br><span class="line">[2,] &quot;ENSMUSG00000000003&quot; &quot;Pbsn&quot;  &quot;0&quot;  &quot;0&quot; </span><br><span class="line">[3,] &quot;ENSMUSG00000000028&quot; &quot;Cdc45&quot; &quot;1&quot;  &quot;0&quot; </span><br><span class="line">[4,] &quot;ENSMUSG00000000031&quot; &quot;H19&quot;   &quot;0&quot;  &quot;2&quot; </span><br><span class="line">[5,] &quot;ENSMUSG00000000037&quot; &quot;Scml2&quot; &quot;0&quot;  &quot;0&quot; </span><br><span class="line">[6,] &quot;ENSMUSG00000000049&quot; &quot;Apoh&quot;  &quot;54&quot; &quot;33&quot;</span><br><span class="line"># 得到的这个swap_rawdata是一个matrix，如果想要让其变为data frame</span><br><span class="line">&gt; swap_rawdata &lt;- data.frame(swap_rawdata)</span><br><span class="line"># 查看一下是否转化成功</span><br><span class="line">&gt; head(swap_rawdata)</span><br><span class="line">     ensembl_gene_id external_gene_name control_W55 test_K54</span><br><span class="line">1 ENSMUSG00000000001              Gnai3           7       11</span><br><span class="line">2 ENSMUSG00000000003               Pbsn           0        0</span><br><span class="line">3 ENSMUSG00000000028              Cdc45           1        0</span><br><span class="line">4 ENSMUSG00000000031                H19           0        2</span><br><span class="line">5 ENSMUSG00000000037              Scml2           0        0</span><br><span class="line">6 ENSMUSG00000000049               Apoh          54       33</span><br><span class="line"># 转化完转化先存一份csv文件在电脑里，便于之后用电脑查看</span><br><span class="line">&gt; write.csv(x &#x3D; swap_rawdata,file &#x3D; &quot;C:&#x2F;&#x2F;Users&#x2F;My&#x2F;Desktop&#x2F;swap_rawdata.csv&quot;)</span><br><span class="line"># 存完之后直接从电脑导入你刚存的文件，这样做可以避免出现numeric数据框变成factor形式</span><br><span class="line">&gt; swap_rawdata &lt;- read.table(&quot;swap_rawdata.csv&quot;,header &#x3D; T,sep &#x3D; &quot;,&quot;)</span><br><span class="line"># 查看</span><br><span class="line">&gt; head(swap_rawdata)</span><br><span class="line">  X    ensembl_gene_id external_gene_name control_W55 test_K54</span><br><span class="line">1 1 ENSMUSG00000000001              Gnai3           7       11</span><br><span class="line">2 2 ENSMUSG00000000003               Pbsn           0        0</span><br><span class="line">3 3 ENSMUSG00000000028              Cdc45           1        0</span><br><span class="line">4 4 ENSMUSG00000000031                H19           0        2</span><br><span class="line">5 5 ENSMUSG00000000037              Scml2           0        0</span><br><span class="line">6 6 ENSMUSG00000000049               Apoh          54       33</span><br><span class="line">&gt; data.class(swap_rawdata[1,1])</span><br></pre></td></tr></table></figure>


<h3 id="2-2-接着构建DGEList对象"><a href="#2-2-接着构建DGEList对象" class="headerlink" title="2.2 接着构建DGEList对象"></a>2.2 接着构建DGEList对象</h3><p>这里因为已经有rawdata的count文件，因此直接用DGEList()函数就行了，否则要用readDGE()函数</p>
<figure class="highlight plain"><table><tr><td class="code"><pre><span class="line"># 首先载入edgeR 包</span><br><span class="line">&gt; library(edgeR)</span><br><span class="line"># 构建DGEList</span><br><span class="line">&gt; group &lt;- 1:2</span><br><span class="line">&gt; y &lt;- DGEList(counts &#x3D; swap_rawdata[,4:5],genes &#x3D; swap_rawdata[,2:3],group &#x3D; group)</span><br><span class="line"># 查看构建完y的信息</span><br><span class="line">&gt; y</span><br></pre></td></tr></table></figure>


<p><strong>查看构建DGElist的运行结果:</strong></p>
<blockquote>
<p> DGEList对象主要有三部分：</p>
</blockquote>
<blockquote>
<p><strong>1、counts矩阵</strong>：包含的是整数counts;</p>
</blockquote>
<blockquote>
<p><strong>2、samples数据框：</strong>包含的是文库(sample)信息。包含 lib.size列 ：for the library size (sequencing depth) for each sample,如果不自定义，  the library sizes will be computed from the column sums of the counts。其中还有一个group列，用于指定每个sample组信息</p>
</blockquote>
<blockquote>
<p>3、<strong>一个可选的数据框genes</strong>：gene的注释信息</p>
</blockquote>
<p><img src="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/6_1.jpg"></p>
<p><strong>第二步： 过滤 low counts数据。</strong>与DESeq2的预过滤不同，DESeq2的预过滤只是为了改善后续运算性能，在运行过程中依旧会自动处理low count数据，edgeR需要在分析前就要排除那些low count数据，而且非常严格。从生物学角度，有生物学意义的基因的表达量必须高于某一个阈值。从统计学角度上， low count的数据不太可能有显著性差异，而且在多重试验矫正阶段还会拖后腿。 </p>
<h5 id="数据过滤"><a href="#数据过滤" class="headerlink" title="数据过滤"></a>数据过滤</h5><p>由于原来的表达量矩阵基因数太大, 可能存在某些基因根本没有表达, 因此需要预先过滤</p>
<figure class="highlight plain"><table><tr><td class="code"><pre><span class="line">&gt; keep &lt;- rowSums(cpm(y)&gt;1) &gt;&#x3D; 1</span><br><span class="line">&gt; y &lt;- y[keep, , keep.lib.sizes&#x3D;FALSE]</span><br><span class="line">&gt; y</span><br></pre></td></tr></table></figure>
<p>这部分代码的意思指的是保留在至少在一个样本里有表达的基因(CPM &gt; 1)。 基因数就从原来的55318变为15868</p>
<p><img src="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/6_2.jpg"></p>
<p>###标准化<br>考虑到测序深度不同, 我们需要对其进行标准化, 避免文库大小不同导致的分析误差.</p>
<p>edgeR里默认采用TMM(trimmed mean of M-values) 对配对样本进行标准化，用到的函数是calcNormFactors</p>
<figure class="highlight plain"><table><tr><td class="code"><pre><span class="line">&gt; y &lt;- calcNormFactors(y)</span><br></pre></td></tr></table></figure>
<p>执行结果：</p>
<p><img src="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/6_3.jpg"></p>
<h3 id="差异表达分析"><a href="#差异表达分析" class="headerlink" title="差异表达分析"></a>差异表达分析</h3><p>不同差异表达分析工具的目标就是预测出dispersion(离散值), 有了离散值就能够计算p值. 那么dispersion怎么计算呢? edgeR给了几个方法</p>
<p>根据经验给定一个值(BCV, square-root-dispersion). edgeR给的建议是, 如果你是人类数据, 且实验做的很好(无过多的其他因素影响), 设置为0.4, 如果是遗传上相似的模式物种（这里为小鼠）, 设置为0.1 （查询edgeR的bioconductor包所得）<br>Simply pick a reasonable dispersion value, based on your experience with similar data,and use that for exactTest or glmFit. Typical values for the common BCV (square-root dispersion) for datasets arising from well-controlled experiments are 0.4 for human data,0.1 for data on genetically identical model organisms or 0.01 for technical replicates.</p>
<figure class="highlight plain"><table><tr><td class="code"><pre><span class="line">&gt; y_bcv &lt;- y</span><br><span class="line"># 因为本次的数据使用的是小鼠的数据，所以使用0.1</span><br><span class="line">&gt; bcv &lt;- 0.1</span><br><span class="line">&gt; et &lt;- exactTest(y_bcv, dispersion &#x3D; bcv ^ 2)</span><br><span class="line">&gt; gene1 &lt;- decideTestsDGE(et, p.value &#x3D; 0.05, lfc &#x3D; 0)</span><br><span class="line">&gt; head(gene1)</span><br><span class="line">&gt; summary(gene1)</span><br></pre></td></tr></table></figure>
<p>执行结果：</p>
<p><img src="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/6_4.jpg"></p>
<p>由统计结果可知，下调的基因为816个，上调的基因为572个</p>
<p>如果觉得觉得基因较多的话，可以上调bcv的值</p>
<figure class="highlight plain"><table><tr><td class="code"><pre><span class="line">&gt; y_bcv &lt;- y</span><br><span class="line">&gt; bcv &lt;- 0.2</span><br><span class="line">&gt; et2 &lt;- exactTest(y_bcv, dispersion &#x3D; bcv ^ 2)</span><br><span class="line">&gt; gene2 &lt;- decideTestsDGE(et2, p.value &#x3D; 0.05, lfc &#x3D; 0)</span><br><span class="line">&gt; summary(gene2)</span><br></pre></td></tr></table></figure>
<p>执行结果：</p>
<p><img src="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/6_5.jpg"></p>
<p>由统计结果可知，下调的基因为377个，上调的基因为221个<br>与之前的结果相比，的确已经减少很多</p>
<p>将结果整理成excel表</p>
<figure class="highlight plain"><table><tr><td class="code"><pre><span class="line"># 改一下gene1的名称</span><br><span class="line">&gt; colnames(gene1) &lt;- &quot;Signifi&quot;</span><br><span class="line"># 组合将所需要的数据组成一个新的data.frame</span><br><span class="line">&gt; results &lt;- cbind(y$genes,y$counts,et$table,gene1)</span><br><span class="line">&gt; head(results)</span><br><span class="line"># 将新生成的results数据框写成一个excel数据表</span><br><span class="line">&gt; write.csv(x &#x3D; results,file &#x3D; &quot;C:&#x2F;&#x2F;Users&#x2F;My&#x2F;Desktop&#x2F;DEresult.csv&quot;,row.names &#x3D; F)</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p>执行结果：</p>
<p><img src="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/6_6.jpg" alt="组合成的新的data.frame表"></p>
<p>生成的excel表可以将down expressed 和 up expressed基因分开</p>
<p><img src="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/6_7.jpg" alt="生成的excel"></p>
<p>参考文献：<a target="_blank" rel="noopener" href="https://blog.csdn.net/u012110870/article/details/102804557">https://blog.csdn.net/u012110870/article/details/102804557</a><br><a target="_blank" rel="noopener" href="https://www.bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf">https://www.bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf</a></p>
<p><a target="_blank" rel="noopener" href="https://cdn.jsdelivr.net/">https://cdn.jsdelivr.net</a></p>

    </div>

    
    
    

    <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/RNAseq/" rel="tag"># RNAseq</a>
          </div>

        

          <div class="post-nav">
            <div class="post-nav-item">
                <a href="/2020/11/23/RNAseq-5-count%E4%B8%8E%E5%9F%BA%E5%9B%A0%E6%B3%A8%E9%87%8A/" rel="prev" title="RNAseq_5_count与基因注释">
                  <i class="fa fa-chevron-left"></i> RNAseq_5_count与基因注释
                </a>
            </div>
            <div class="post-nav-item">
                <a href="/2020/12/01/%E6%95%B0%E6%8D%AE%E6%95%B4%E7%90%86%E4%B8%8Etidyverse%E7%B3%BB%E7%BB%9F/" rel="next" title="数据整理与tidyverse系统">
                  数据整理与tidyverse系统 <i class="fa fa-chevron-right"></i>
                </a>
            </div>
          </div>
    </footer>
  </article>
  
  
  



      
    <div class="comments" id="valine-comments"></div>

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      const activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      const commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

    </div>
  </main>

  <footer class="footer">
    <div class="footer-inner">
      

      

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2020</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">WENJU QIAN</span>
</div>
<div class="busuanzi-count">
    <span class="post-meta-item" id="busuanzi_container_site_uv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="fa fa-user"></i>
      </span>
      <span class="site-uv" title="Total Visitors">
        <span id="busuanzi_value_site_uv"></span>
      </span>
    </span>
    <span class="post-meta-item" id="busuanzi_container_site_pv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="fa fa-eye"></i>
      </span>
      <span class="site-pv" title="Total Views">
        <span id="busuanzi_value_site_pv"></span>
      </span>
    </span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> & <a href="https://theme-next.js.org/" class="theme-link" rel="noopener" target="_blank">NexT.Gemini</a>
  </div>

    </div>
  </footer>

  
  <script src="//cdn.jsdelivr.net/npm/animejs@3.2.0/lib/anime.min.js"></script>
<script src="/js/utils.js"></script><script src="/js/motion.js"></script><script src="/js/next-boot.js"></script>

  




  <script src="/js/local-search.js"></script>












  
  <script async src="https://busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>








  

  
<script>
NexT.utils.loadComments('#valine-comments', () => {
  NexT.utils.getScript('//cdn.jsdelivr.net/npm/valine@1.4.14/dist/Valine.min.js', () => {
    new Valine(Object.assign({
      el  : '#valine-comments',
      path: "/2020/11/23/RNAseq-6-%E6%97%A0%E9%87%8D%E5%A4%8D%E5%B7%AE%E5%BC%82%E5%9F%BA%E5%9B%A0%E5%88%86%E6%9E%90/",
    }, {"enable":true,"appId":"0uYeC1X5ME3X6vtooIJoYito-gzGzoHsz","appKey":"7u586HL29S9uVvurXaAzOhfn","placeholder":"Just go go","avatar":"mm","meta":["nick","mail","link"],"pageSize":10,"lang":null,"visitor":false,"comment_count":true,"recordIP":false,"serverURLs":null,"enableQQ":false,"requiredFields":[]}
    ));
  }, window.Valine);
});
</script>


<!-- <script type="text/javascript" color="0,0,0" opacity='0.5' zIndex="-1" count="99" src="//cdn.bootcss.com/canvas-nest.js/1.0.0/canvas-nest.min.js"></script> -->


</body>
</html>
