<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.2.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">



<link rel="stylesheet" href="//cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.14.0/css/all.min.css">
  <link rel="stylesheet" href="//cdn.jsdelivr.net/npm/animate.css@3.1.1/animate.min.css">


<script class="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"example.com","root":"/","scheme":"Gemini","version":"8.0.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12},"copycode":false,"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"fadeInDown","post_body":"fadeInDown","coll_header":"fadeInLeft","sidebar":"fadeInUp"}},"prism":false,"i18n":{"placeholder":"Searching...","empty":"We didn't find any results for the search: ${query}","hits_time":"${hits} results found in ${time} ms","hits":"${hits} results found"},"path":"search.xml","localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false}};
  </script>

  <meta name="description" content="理论基础在上篇的比对中，我们需要纠结是否真的需要比对，如果你只需要知道已知基因的表达情况，那么可以选择alignment-free工具（例如salmon, sailfish)，如果你需要找到noval isoforms，那么就需要alignment-based工具（如HISAT2, STAR）。到了这一篇的基因（转录本）定量，需要考虑的因素就更加多了，以至于我不知道如何说清才能理清逻辑。 定量分为">
<meta property="og:type" content="article">
<meta property="og:title" content="RNAseq_5_count与基因注释">
<meta property="og:url" content="http://example.com/2020/11/23/RNAseq-5-count%E4%B8%8E%E5%9F%BA%E5%9B%A0%E6%B3%A8%E9%87%8A/index.html">
<meta property="og:site_name" content="WENJU QIAN‘s tech blog">
<meta property="og:description" content="理论基础在上篇的比对中，我们需要纠结是否真的需要比对，如果你只需要知道已知基因的表达情况，那么可以选择alignment-free工具（例如salmon, sailfish)，如果你需要找到noval isoforms，那么就需要alignment-based工具（如HISAT2, STAR）。到了这一篇的基因（转录本）定量，需要考虑的因素就更加多了，以至于我不知道如何说清才能理清逻辑。 定量分为">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_1.jpg">
<meta property="og:image" content="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_2.jpg">
<meta property="og:image" content="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_3.jpg">
<meta property="og:image" content="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_4.jpg">
<meta property="og:image" content="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_5.jpg">
<meta property="og:image" content="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_6.jpg">
<meta property="og:image" content="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_7.jpg">
<meta property="og:image" content="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_8.jpg">
<meta property="og:image" content="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_9.jpg">
<meta property="og:image" content="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_10.jpg">
<meta property="og:image" content="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_11.jpg">
<meta property="og:image" content="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_12.jpg">
<meta property="og:image" content="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_13.jpg">
<meta property="og:image" content="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_14.jpg">
<meta property="article:published_time" content="2020-11-22T16:22:00.000Z">
<meta property="article:modified_time" content="2020-11-30T15:53:14.819Z">
<meta property="article:author" content="WENJU QIAN">
<meta property="article:tag" content="RNAseq">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_1.jpg">


<link rel="canonical" href="http://example.com/2020/11/23/RNAseq-5-count%E4%B8%8E%E5%9F%BA%E5%9B%A0%E6%B3%A8%E9%87%8A/">


<script class="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'en'
  };
</script>

  <title>RNAseq_5_count与基因注释 | WENJU QIAN‘s tech blog</title>
  






  <noscript>
  <style>
  body { margin-top: 2rem; }

  .use-motion .menu-item,
  .use-motion .sidebar,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header {
    visibility: visible;
  }

  .use-motion .header,
  .use-motion .site-brand-container .toggle,
  .use-motion .footer { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle,
  .use-motion .custom-logo-image {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line {
    transform: scaleX(1);
  }

  .search-pop-overlay, .sidebar-nav { display: none; }
  .sidebar-panel { display: block; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage" class="use-motion">
  <div class="headband"></div>

  <main class="main">
    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
        <span class="toggle-line"></span>
        <span class="toggle-line"></span>
        <span class="toggle-line"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <i class="logo-line"></i>
      <h1 class="site-title">WENJU QIAN‘s tech blog</h1>
      <i class="logo-line"></i>
    </a>
      <p class="site-subtitle" itemprop="description">越努力，越幸运~</p>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>



<nav class="site-nav">
  <ul class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="/about/" rel="section"><i class="fa fa-user fa-fw"></i>About</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/tags/" rel="section"><i class="fa fa-tags fa-fw"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/categories/" rel="section"><i class="fa fa-th fa-fw"></i>Categories</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a>

  </li>
        <li class="menu-item menu-item-schedule">

    <a href="/schedule/" rel="section"><i class="fa fa-calendar fa-fw"></i>Schedule</a>

  </li>
        <li class="menu-item menu-item-sitemap">

    <a href="/sitemap.xml" rel="section"><i class="fa fa-sitemap fa-fw"></i>Sitemap</a>

  </li>
        <li class="menu-item menu-item-commonweal">

    <a href="/404/" rel="section"><i class="fa fa-heartbeat fa-fw"></i>Commonweal 404</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off" maxlength="80"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div class="search-result-container no-result">
  <div class="search-result-icon">
    <i class="fa fa-spinner fa-pulse fa-5x"></i>
  </div>
</div>

    </div>
  </div>

</div>
        
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line"></span>
    <span class="toggle-line"></span>
    <span class="toggle-line"></span>
  </div>

  <aside class="sidebar">

    <div class="sidebar-inner sidebar-nav-active sidebar-toc-active">
      <ul class="sidebar-nav">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <section class="post-toc-wrap sidebar-panel">
          <div class="post-toc animated"><ol class="nav"><li class="nav-item nav-level-2"><a class="nav-link" href="#%E7%90%86%E8%AE%BA%E5%9F%BA%E7%A1%80"><span class="nav-number">1.</span> <span class="nav-text">理论基础</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#%E5%B0%8F%E7%BB%93"><span class="nav-number">1.1.</span> <span class="nav-text">小结</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E8%BE%93%E5%87%BA%E8%A1%A8%E8%BE%BE%E7%9F%A9%E9%98%B5"><span class="nav-number">2.</span> <span class="nav-text">输出表达矩阵</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#%E5%90%88%E5%B9%B6%E8%A1%A8%E8%BE%BE%E7%9F%A9%E9%98%B5%E5%B9%B6%E8%BF%9B%E8%A1%8C%E6%B3%A8%E9%87%8A%EF%BC%88R%E4%B8%AD%E8%BF%9B%E8%A1%8C%EF%BC%89"><span class="nav-number">2.0.1.</span> <span class="nav-text">合并表达矩阵并进行注释（R中进行）</span></a><ol class="nav-child"><li class="nav-item nav-level-6"><a class="nav-link" href="#%E4%BB%8E%E4%B8%8A%E9%9D%A2%E7%9C%8B%E5%87%BA%E9%9C%80%E8%A6%81%E8%87%B3%E5%B0%91%E5%81%9A%E4%B8%A4%E6%AD%A5%E5%B7%A5%E4%BD%9C%E6%89%8D%E8%83%BD%E6%9B%B4%E5%A5%BD%E7%90%86%E8%A7%A3%E5%92%8C%E5%BE%80%E4%B8%8B%E8%BF%9B%E8%A1%8C%E5%88%86%E6%9E%90"><span class="nav-number">2.0.1.0.1.</span> <span class="nav-text">从上面看出需要至少做两步工作才能更好理解和往下进行分析</span></a></li><li class="nav-item nav-level-6"><a class="nav-link" href="#%E7%AC%AC%E4%B8%80%EF%BC%8C%E9%9C%80%E8%A6%81%E6%8A%8A2%E4%B8%AA%E6%96%87%E4%BB%B6%E5%90%88%E5%B9%B6%EF%BC%9B"><span class="nav-number">2.0.1.0.2.</span> <span class="nav-text">第一，需要把2个文件合并；</span></a></li><li class="nav-item nav-level-6"><a class="nav-link" href="#%E7%AC%AC%E4%BA%8C%EF%BC%8C%E9%9C%80%E8%A6%81%E6%8A%8Aensembl-gene-id%E8%BD%AC%E6%8D%A2%E4%B8%BAgene-symbol-%EF%BC%88%E8%BF%99%E4%B8%80%E6%AD%A5%E4%B8%8D%E8%BF%9B%E8%A1%8C%E4%B9%9F%E8%A1%8C%EF%BC%8C%E5%90%8E%E9%9D%A2%E8%BF%98%E9%9C%80%E8%A6%81%EF%BC%89"><span class="nav-number">2.0.1.0.3.</span> <span class="nav-text">第二，需要把ensembl_gene_id转换为gene_symbol;（这一步不进行也行，后面还需要）</span></a></li><li class="nav-item nav-level-6"><a class="nav-link" href="#1-%E8%BD%BD%E5%85%A5%E6%95%B0%E6%8D%AE%EF%BC%8C%E6%B7%BB%E5%8A%A0%E5%88%97%E5%90%8D"><span class="nav-number">2.0.1.0.4.</span> <span class="nav-text">(1) 载入数据，添加列名</span></a></li><li class="nav-item nav-level-6"><a class="nav-link" href="#3-%E5%AF%B9%E5%9F%BA%E5%9B%A0%E8%BF%9B%E8%A1%8C%E6%B3%A8%E9%87%8A-%E8%8E%B7%E5%8F%96gene-symbol"><span class="nav-number">2.0.1.0.5.</span> <span class="nav-text">(3) 对基因进行注释-获取gene_symbol</span></a></li><li class="nav-item nav-level-6"><a class="nav-link" href="#%E4%BB%A5%E4%B8%8B%E4%B8%A4%E7%A7%8D%E6%96%B9%E5%BC%8F%E5%8F%AF%E4%BB%A5%E8%BF%9B%E8%A1%8C"><span class="nav-number">2.0.1.0.6.</span> <span class="nav-text">以下两种方式可以进行</span></a></li></ol></li></ol></li></ol></li></ol></li></ol></div>
      </section>
      <!--/noindex-->

      <section class="site-overview-wrap sidebar-panel">
        <div class="site-author animated" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="WENJU QIAN"
      src="/images/avatar.jpg">
  <p class="site-author-name" itemprop="name">WENJU QIAN</p>
  <div class="site-description" itemprop="description">Student From Shanghai Institute of Nutrition and Health, CAS</div>
</div>
<div class="site-state-wrap animated">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">19</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
          
        <span class="site-state-item-count">4</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author animated">
      <span class="links-of-author-item">
        <a href="https://github.com/qianwenju" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;qianwenju" rel="noopener" target="_blank"><i class="fab fa-github fa-fw"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:qianwenju2018@sibs.ac.cn" title="E-Mail → mailto:qianwenju2018@sibs.ac.cn" rel="noopener" target="_blank"><i class="fa fa-envelope fa-fw"></i>E-Mail</a>
      </span>
  </div>



      </section>
    </div>
  </aside>
  <div class="sidebar-dimmer"></div>
 


    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>

<noscript>
  <div class="noscript-warning">Theme NexT works best with JavaScript enabled</div>
</noscript>


    <div class="main-inner post posts-expand">
      

      

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="http://example.com/2020/11/23/RNAseq-5-count%E4%B8%8E%E5%9F%BA%E5%9B%A0%E6%B3%A8%E9%87%8A/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.jpg">
      <meta itemprop="name" content="WENJU QIAN">
      <meta itemprop="description" content="Student From Shanghai Institute of Nutrition and Health, CAS">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="WENJU QIAN‘s tech blog">
    </span>

    
    
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          RNAseq_5_count与基因注释
        </h1>

        <div class="post-meta-container">
          <div class="post-meta">
    <span class="post-meta-item">
      <span class="post-meta-item-icon">
        <i class="far fa-calendar"></i>
      </span>
      <span class="post-meta-item-text">Posted on</span>

      <time title="Created: 2020-11-23 00:22:00" itemprop="dateCreated datePublished" datetime="2020-11-23T00:22:00+08:00">2020-11-23</time>
    </span>
      <span class="post-meta-item">
        <span class="post-meta-item-icon">
          <i class="far fa-calendar-check"></i>
        </span>
        <span class="post-meta-item-text">Edited on</span>
        <time title="Modified: 2020-11-30 23:53:14" itemprop="dateModified" datetime="2020-11-30T23:53:14+08:00">2020-11-30</time>
      </span>

  
  
  <span class="post-meta-item">
    
      <span class="post-meta-item-icon">
        <i class="far fa-comment"></i>
      </span>
      <span class="post-meta-item-text">Valine: </span>
    
    <a title="valine" href="/2020/11/23/RNAseq-5-count%E4%B8%8E%E5%9F%BA%E5%9B%A0%E6%B3%A8%E9%87%8A/#valine-comments" itemprop="discussionUrl">
      <span class="post-comments-count valine-comment-count" data-xid="/2020/11/23/RNAseq-5-count%E4%B8%8E%E5%9F%BA%E5%9B%A0%E6%B3%A8%E9%87%8A/" itemprop="commentCount"></span>
    </a>
  </span>
  
  
</div>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">
        <h2 id="理论基础"><a href="#理论基础" class="headerlink" title="理论基础"></a>理论基础</h2><p>在上篇的比对中，我们需要纠结是否真的需要比对，如果你只需要知道已知基因的表达情况，那么可以选择alignment-free工具（例如salmon, sailfish)，如果你需要找到noval isoforms，那么就需要alignment-based工具（如HISAT2, STAR）。到了这一篇的基因（转录本）定量，需要考虑的因素就更加多了，以至于我不知道如何说清才能理清逻辑。</p>
<p>定量分为三个水平</p>
<ul>
<li>  基因水平(gene-level)</li>
<li>  转录本水平(transcript-level)</li>
<li>  外显子使用水平(exon-usage-level)。</li>
</ul>
<p>在<strong>基因水平</strong>上，常用的软件为HTSeq-count，featureCounts，BEDTools, Qualimap, Rsubread, GenomicRanges等。以常用的HTSeq-count为例，这些工具要解决的问题就是根据read和基因位置的overlap判断这个read到底是谁家的孩子。值得<strong>注意</strong>的是不同工具对multimapping reads处理方式也是不同的，例如HTSeq-count就直接当它们不存在。而Qualimpa则是一人一份，平均分配。</p>
<p><img src="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_1.jpg" alt="image"></p>
<p>对每个基因计数之后得到的count matrix再后续的分析中，要注意标准化的问题。如果你要比较同一个样本(within-sample)不同基因之间的表达情况，你就需要考虑到<strong>转录本长度</strong>，因为转录本越长，那么检测的片段也会更多，直接比较等于让小孩和大人进行赛跑。如果你是比较不同样本（across sample）同一个基因的表达情况，虽然不必在意转录本长度，但是你要考虑到<strong>测序深度</strong>（sequence depth)，毕竟测序深度越高，检测到的概率越大。除了这两个因素外，你还需要考虑GC%所导致的偏差，以及测序仪器的系统偏差。目前对read count标准化的算法有RPKM（SE）, FPKM（PE），TPM, TMM等，不同算法之间的差异与换算方法已经有文章进行整理和吐槽了。<strong>但是</strong>，有一些下游分析的软件会要求是输入的count matrix是原始数据，未经标准化，比如说DESeq2，这个时候你需要<strong>注意</strong>你上一步所用软件会不会进行标准化。</p>
<p>在<strong>转录本水平</strong>上，一般常用工具为Cufflinks和它的继任者StringTie， eXpress。这些软件要处理的难题就时转录本亚型（isoforms）之间通常是有重叠的，当二代测序读长低于转录本长度时，如何进行区分？这些工具大多采用的都是expectation maximization（EM）。好在我们有三代测序。</p>
<p>上述软件都是alignment-based，目前许多alignment-free软件，如kallisto, silfish, salmon，能够省去比对这一步，直接得到read count，在运行效率上更高。不过最近一篇文献[1]指出这类方法在估计丰度时存在样本特异性和读长偏差。</p>
<p>在<strong>外显子使用水平</strong>上，其实和基因水平的统计类似。但是值得注意的是为了更好的计数，我们需要提供无重叠的外显子区域的gtf文件[2]。用于分析差异外显子使用的DEXSeq提供了一个Python脚本（dexseq_prepare_annotation.py）执行这个任务。</p>
<h3 id="小结"><a href="#小结" class="headerlink" title="小结"></a>小结</h3><p>计数分为三个水平： gene-level, transcript-level, exon-usage-level</p>
<p>标准化方法： FPKM RPKM TMM TPM</p>
<h2 id="输出表达矩阵"><a href="#输出表达矩阵" class="headerlink" title="输出表达矩阵"></a>输出表达矩阵</h2><p>在RNA-Seq分析中，每一个基因就是一个feature（特征？），而基因被认为是它的所有外显子的和集。在可变剪切分析中，可以单独把每个外显子当作一个feature。而在ChIP-Seq分析中，feature则是预先定义的结合域。但是确定一个read到底属于哪一个feature有时会非常棘手。因此HTSeq提供了三种模式，示意图见前一幅图</p>
<ul>
<li>  the union of all the sets S(i) for mode union. This mode is recommended for most use cases.</li>
<li>  the intersection of all the sets S(i) for mode intersection-strict.</li>
<li>  the intersection of all non-empty sets S(i) for mode intersection-nonempty.</li>
</ul>
<p>基本用法非常的简单：</p>
<blockquote>
<p>htseq-count [options] <alignment_file> <gtf_file></p>
</blockquote>
<figure class="highlight plain"><table><tr><td class="code"><pre><span class="line"># 安装</span><br><span class="line">conda install htseq</span><br><span class="line"># 利用htseq-count对sort之后的bam文件进行reads计数</span><br><span class="line">htseq-count -r pos -f bam &#x2F;home&#x2F;cenhui2018&#x2F;QWJ&#x2F;sequence_data&#x2F;20191030_NGS_DATA&#x2F;19R577_paired.hisat2_sorted.bam &#x2F;home&#x2F;cenhui2018&#x2F;QWJ&#x2F;sequence_data&#x2F;genome&#x2F;Anotation&#x2F;gencode.vM23.annotation.gtf &gt; 19R577_paired.count 2&gt;&#x2F;home&#x2F;cenhui2018&#x2F;QWJ&#x2F;sequence_data&#x2F;20191030_NGS_DATA&#x2F;19R577_htseq.log &amp;</span><br><span class="line"></span><br></pre></td></tr></table></figure>

<p>运行的时间会比较久，所以可以去了解不同参数的用法了，其中比较常用的为：</p>
<ul>
<li>  -f bam/sam： 指定输入文件格式，默认SAM</li>
<li>  -r name/pos: 你需要利用samtool sort对数据根据read name或者位置进行排序，默认是name</li>
<li>  -s yes/no/reverse: 数据是否来自于strand-specific assay。DNA是双链的，所以需要判断到底来自于哪条链。如果选择了no， 那么每一条read都会跟正义链和反义链进行比较。默认的yes对于双端测序表示第一个read都在同一个链上，第二个read则在另一条链上。</li>
<li>  -a 最低质量， 剔除低于阈值的read</li>
<li>  -m 模式 union（默认）, intersection-strict and intersection-nonempty。一般而言就用默认的，作者也是这样认为的。</li>
<li>  -i id attribute: 在GTF文件的最后一栏里，会有这个基因的多个命名方式（如下）， RNA-Seq数据分析常用的是gene_id， 当然你可以写一个脚本替换成其他命名方式。</li>
</ul>
<p>对count数据进行查看</p>
<figure class="highlight plain"><table><tr><td class="code"><pre><span class="line"># 利用wc指令我们可以计算count文件的行数</span><br><span class="line">&gt; wc -l *.count</span><br></pre></td></tr></table></figure>

<p>运行结果：</p>
<p><img src="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_2.jpg" alt="统计count文件的行数"></p>
<p>看下每个文件的格式，</p>
<figure class="highlight plain"><table><tr><td class="code"><pre><span class="line"># 查看前4行,第一列ensembl_gene_id,第二列read_count计数</span><br><span class="line">&gt; $ head -n 4 *.count     </span><br><span class="line"> # 查看后四行</span><br><span class="line">&gt; $ tail -n 4 *.count</span><br></pre></td></tr></table></figure>

<p>运行结果：</p>
<p><img src="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_3.jpg" alt="查看count文件的首行及末行"></p>
<h4 id="合并表达矩阵并进行注释（R中进行）"><a href="#合并表达矩阵并进行注释（R中进行）" class="headerlink" title="合并表达矩阵并进行注释（R中进行）"></a>合并表达矩阵并进行注释（R中进行）</h4><p>上一步得到的2个单独的矩阵文件，现在要把这<strong>2个文件合并为行为基因名，列为样本名，中间为count的矩阵文件</strong>。</p>
<ul>
<li><h6 id="从上面看出需要至少做两步工作才能更好理解和往下进行分析"><a href="#从上面看出需要至少做两步工作才能更好理解和往下进行分析" class="headerlink" title="从上面看出需要至少做两步工作才能更好理解和往下进行分析"></a>从上面看出需要至少做两步工作才能更好理解和往下进行分析</h6></li>
</ul>
<h6 id="第一，需要把2个文件合并；"><a href="#第一，需要把2个文件合并；" class="headerlink" title="第一，需要把2个文件合并；"></a>第一，需要把2个文件合并；</h6><h6 id="第二，需要把ensembl-gene-id转换为gene-symbol-（这一步不进行也行，后面还需要）"><a href="#第二，需要把ensembl-gene-id转换为gene-symbol-（这一步不进行也行，后面还需要）" class="headerlink" title="第二，需要把ensembl_gene_id转换为gene_symbol;（这一步不进行也行，后面还需要）"></a>第二，需要把ensembl_gene_id转换为gene_symbol;（这一步不进行也行，后面还需要）</h6><ul>
<li>  熟悉python的朋友可以参考<a target="_blank" rel="noopener" href="https://www.jianshu.com/p/e9742bbf83b9">这篇文章</a></li>
<li>  我用R中的merge命令来处理，参考<a target="_blank" rel="noopener" href="https://www.jianshu.com/p/24cf44b610a7">这里</a>和<a target="_blank" rel="noopener" href="https://www.jianshu.com/p/e9742bbf83b9">这里</a></li>
</ul>
<p>运行R设置工作路径并查看当前目录下的文件</p>
<figure class="highlight plain"><table><tr><td class="code"><pre><span class="line">&gt; setwd(&quot;&#x2F;home&#x2F;cenhui2018&#x2F;QWJ&#x2F;sequence_data&#x2F;20191030_NGS_DATA&#x2F;&quot;)</span><br><span class="line">&gt; list.files()</span><br></pre></td></tr></table></figure>
<p>运行结果</p>
<p><img src="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_4.jpg"></p>
<h6 id="1-载入数据，添加列名"><a href="#1-载入数据，添加列名" class="headerlink" title="(1) 载入数据，添加列名"></a>(1) 载入数据，添加列名</h6><figure class="highlight plain"><table><tr><td class="code"><pre><span class="line"># 防止R自动把字符串string的列辨认成factor</span><br><span class="line">&gt; options(stringAsFactor &#x3D;  FALSE)  </span><br><span class="line"># 从19R576_paired.count文件中读取数据，并添加列名，之后生成新文件命名为control_W55</span><br><span class="line">&gt; control_W55 &lt;- read.table(&quot;19R576_paired.count&quot;,sep &#x3D; &quot;\t&quot;,col.names&#x3D;c(&quot;gene_id&quot;,&quot;control_W55&quot;))</span><br><span class="line"># 查看前6行</span><br><span class="line">&gt; head(control_W55)</span><br><span class="line"># 类似的方法处理19R577_paired.count，并存储到</span><br><span class="line">&gt; test_K54 &lt;- read.table(&quot;19R577_paired.count&quot;,sep &#x3D; &quot;\t&quot;,col.names&#x3D;c(&quot;gene_id&quot;,&quot;test_K54&quot;))</span><br></pre></td></tr></table></figure>
<p>运行结果：</p>
<p><img src="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_5.jpg"></p>
<p>######(2)数据整合<br><strong>merge进行整合</strong><br>gencode的注释文件中的gene_id（如ENSMUSG00000105298.13_3）在EBI是不能搜索到的，所以用gsub功能只保留ENSMUSG00000105298这部分<br>处理之前先看一下,也就是最后5行是我们不需要的，可以删除</p>
<p><img src="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_6.jpg" alt="查看文件的最后五行"></p>
<p>整合文件：</p>
<figure class="highlight plain"><table><tr><td class="code"><pre><span class="line"># 将control_w55和test_K54这两个文件通过&quot;gene_id&quot;进行merge，生成新的merge好的文件raw_count</span><br><span class="line">&gt; raw_count &lt;- merge(control_W55,test_K54, by &#x3D; &quot;gene_id&quot;)</span><br><span class="line"># 查看merge后的文件的前6行</span><br><span class="line">&gt; head(raw_count)</span><br><span class="line"># 查看后六行</span><br><span class="line">&gt; tail(raw_count)</span><br></pre></td></tr></table></figure>
<p>运行结果：</p>
<p><img src="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_7.jpg" alt="合并文件并查看前六行"><br><img src="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_8.jpg" alt="查看合并后的文件的后六行"></p>
<p>######删除前5行<br>这里要注意，因为读入之后顺序变了，删除的时候看下删除的是哪些行</p>
<figure class="highlight plain"><table><tr><td class="code"><pre><span class="line">&gt; raw_count_filt &lt;- raw_count[-1:-5,]</span><br><span class="line">&gt; head(raw_count_filt)</span><br></pre></td></tr></table></figure>
<p>删除之后的结果：</p>
<p><img src="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_9.jpg"><br>######因为我们无法在EBI数据库上直接搜索找到ENSMUSG00000024045.5这样的基因，只能是ENSMUSG00000024045的整数，没有小数点，所以需要进一步替换为整数的形式。</p>
<figure class="highlight plain"><table><tr><td class="code"><pre><span class="line"># 第一步通过gsub工具将匹配到的.以及后面的数字连续匹配并替换为空，并赋值给ENSEMBL</span><br><span class="line">&gt;ENSEMBL &lt;- gsub(&quot;\\.\\d*&quot;, &quot;&quot;, raw_count_filt$gene_id) </span><br><span class="line"># 将ENSEMBL重新添加到raw_count_filt1矩阵</span><br><span class="line">&gt;row.names(raw_count_filt) &lt;- ENSEMBL</span><br><span class="line"># 查看&gt;head(raw_count_filt)</span><br></pre></td></tr></table></figure>
<p>执行结果：</p>
<p><img src="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_10.jpg"></p>
<p>可以将最后得到的raw_count_filt文件通过save()函数保存成.Rdata文件，然后复制放在自己的桌面，然后用自己的Rstudio打开</p>
<h6 id="3-对基因进行注释-获取gene-symbol"><a href="#3-对基因进行注释-获取gene-symbol" class="headerlink" title="(3) 对基因进行注释-获取gene_symbol"></a>(3) 对基因进行注释-获取gene_symbol</h6><h6 id="以下两种方式可以进行"><a href="#以下两种方式可以进行" class="headerlink" title="以下两种方式可以进行"></a>以下两种方式可以进行</h6><p>第一：去<a target="_blank" rel="noopener" href="https://biodbnet-abcc.ncifcrf.gov/db/db2db.php">这里</a>或<a target="_blank" rel="noopener" href="https://www.biotools.fr/mouse/ensembl_symbol_converter">这里</a>的网页版，输入列表即可输出，不再赘述<br>第二：用bioMart对ensembl_id转换成gene_symbol<br>bioMart包是一个连接bioMart数据库的R语言接口，能通过这个软件包自由连接到bioMart数据库</p>
<figure class="highlight plain"><table><tr><td class="code"><pre><span class="line"># 首先检查BioManager包是否安装</span><br><span class="line">if (!requireNamespace(&quot;BiocManager&quot;, quietly &#x3D; TRUE))</span><br><span class="line">    install.packages(&quot;BiocManager&quot;)</span><br><span class="line"># 如果BioManager包已经安装，则通过BioManager包安装biomaRt包</span><br><span class="line">BiocManager::install(&quot;biomaRt&quot;)</span><br><span class="line"># 因为我的文件是放在桌面上的，所以先设置R的工作目录为桌面路径</span><br><span class="line">&gt; setwd(&quot;C:&#x2F;&#x2F;Users&#x2F;My&#x2F;Desktop&#x2F;&quot;)</span><br><span class="line"># 加载放在桌面上的.Rdata文件</span><br><span class="line">&gt; load(&quot;raw_count.Rdata&quot;)</span><br><span class="line"># 加载所需要的R包</span><br><span class="line">&gt; library(&#39;biomaRt&#39;)</span><br><span class="line">&gt; library(&quot;curl&quot;)</span><br><span class="line"># 用bioMart对差异表达基因进行注释</span><br><span class="line">&gt; mart &lt;- useDataset(&quot;mmusculus_gene_ensembl&quot;, useMart(&quot;ensembl&quot;))</span><br><span class="line">&gt; my_ensembl_gene_id&lt;-row.names(raw_count_filt)</span><br><span class="line">&gt; mms_symbols&lt;- getBM(attributes&#x3D;c(&#39;ensembl_gene_id&#39;,&#39;external_gene_name&#39;,&quot;description&quot;),filters &#x3D; &#39;ensembl_gene_id&#39;, values &#x3D; my_ensembl_gene_id, mart &#x3D; mart)</span><br></pre></td></tr></table></figure>
<p>执行完之后需要等待一段时间提交检索，下方会有一个进度条</p>
<p><img src="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_11.jpg"></p>
<p>基因注释结束后，查看注释结果</p>
<p><img src="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_12.jpg" alt="基因注释结束后的结果"></p>
<p>####合并数据成一个文件</p>
<p>注意：合并的话两个数据必须有共同的列名，我们先看一下</p>
<p><img src="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_13.jpg"></p>
<p>可见，两个文件没有共同的列名，所以要先给’raw_count_filt’添加一个‘ensembl_gene_id’的列名<br>方法如下:</p>
<figure class="highlight plain"><table><tr><td class="code"><pre><span class="line"># 将raw_count_filt的行名赋值到一个ensembl_gene_id的变量</span><br><span class="line">&gt; ensembl_gene_id&lt;-rownames(raw_count_filt)</span><br><span class="line"># 将行名单独变成一列加到raw_count_filt文件中</span><br><span class="line">&gt; raw_count_filt &lt;- cbind(ensembl_gene_id,raw_count_filt)</span><br><span class="line"># 添加列名</span><br><span class="line">&gt; colnames(raw_count_filt)[1] &lt;- c(&quot;ensembl_gene_id&quot;)</span><br><span class="line"># 将raw_count_filt文件和mms_symbols文件通过ensembl_gene_id而merge在一起，生成新的文件</span><br><span class="line">&gt; diff_name&lt;-merge(raw_count_filt,mms_symbols,by&#x3D;&quot;ensembl_gene_id&quot;)</span><br><span class="line"># 查看新文件前6行</span><br><span class="line">&gt; head(diff_name)</span><br></pre></td></tr></table></figure>
<p>代码执行结果：</p>
<p><img src="https://raw.githubusercontent.com/kevinQIAN1996/Figurebed/main/img/5_14.jpg" alt="执行结果"></p>
<p>现在成功地生成了一个带有基因名称的count表达矩阵</p>
<p>参考链接：<br><a target="_blank" rel="noopener" href="https://blog.csdn.net/lztttao/article/details/82086346">https://blog.csdn.net/lztttao/article/details/82086346</a><br><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/3a0e1e3e41d0">https://www.jianshu.com/p/3a0e1e3e41d0</a></p>

    </div>

    
    
    

    <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/RNAseq/" rel="tag"># RNAseq</a>
          </div>

        

          <div class="post-nav">
            <div class="post-nav-item">
                <a href="/2020/11/23/RNAseq-4-%E5%BA%8F%E5%88%97%E6%AF%94%E5%AF%B9%E4%BB%A5%E5%8F%8Asamtools%E6%A0%BC%E5%BC%8F%E8%BD%AC%E6%8D%A2/" rel="prev" title="RNAseq_4_序列比对以及samtools格式转换">
                  <i class="fa fa-chevron-left"></i> RNAseq_4_序列比对以及samtools格式转换
                </a>
            </div>
            <div class="post-nav-item">
                <a href="/2020/11/23/RNAseq-6-%E6%97%A0%E9%87%8D%E5%A4%8D%E5%B7%AE%E5%BC%82%E5%9F%BA%E5%9B%A0%E5%88%86%E6%9E%90/" rel="next" title="RNAseq_6_无重复差异基因分析">
                  RNAseq_6_无重复差异基因分析 <i class="fa fa-chevron-right"></i>
                </a>
            </div>
          </div>
    </footer>
  </article>
  
  
  



      
    <div class="comments" id="valine-comments"></div>

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      const activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      const commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

    </div>
  </main>

  <footer class="footer">
    <div class="footer-inner">
      

      

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2020</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">WENJU QIAN</span>
</div>
<div class="busuanzi-count">
    <span class="post-meta-item" id="busuanzi_container_site_uv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="fa fa-user"></i>
      </span>
      <span class="site-uv" title="Total Visitors">
        <span id="busuanzi_value_site_uv"></span>
      </span>
    </span>
    <span class="post-meta-item" id="busuanzi_container_site_pv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="fa fa-eye"></i>
      </span>
      <span class="site-pv" title="Total Views">
        <span id="busuanzi_value_site_pv"></span>
      </span>
    </span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> & <a href="https://theme-next.js.org/" class="theme-link" rel="noopener" target="_blank">NexT.Gemini</a>
  </div>

    </div>
  </footer>

  
  <script src="//cdn.jsdelivr.net/npm/animejs@3.2.0/lib/anime.min.js"></script>
<script src="/js/utils.js"></script><script src="/js/motion.js"></script><script src="/js/next-boot.js"></script>

  




  <script src="/js/local-search.js"></script>












  
  <script async src="https://busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>








  

  
<script>
NexT.utils.loadComments('#valine-comments', () => {
  NexT.utils.getScript('//cdn.jsdelivr.net/npm/valine@1.4.14/dist/Valine.min.js', () => {
    new Valine(Object.assign({
      el  : '#valine-comments',
      path: "/2020/11/23/RNAseq-5-count%E4%B8%8E%E5%9F%BA%E5%9B%A0%E6%B3%A8%E9%87%8A/",
    }, {"enable":true,"appId":"0uYeC1X5ME3X6vtooIJoYito-gzGzoHsz","appKey":"7u586HL29S9uVvurXaAzOhfn","placeholder":"Just go go","avatar":"mm","meta":["nick","mail","link"],"pageSize":10,"lang":null,"visitor":false,"comment_count":true,"recordIP":false,"serverURLs":null,"enableQQ":false,"requiredFields":[]}
    ));
  }, window.Valine);
});
</script>


<!-- <script type="text/javascript" color="0,0,0" opacity='0.5' zIndex="-1" count="99" src="//cdn.bootcss.com/canvas-nest.js/1.0.0/canvas-nest.min.js"></script> -->


</body>
</html>
